yoyodyne/sizing.py

```diff
@@ -87,53 +54,68 @@ def find_batch_size(
     model: models.BaseEncoderDecoder,
     datamodule: data.DataModule,
     *,
-    max_trials: int = defaults.FIND_BATCH_SIZE_MAX_TRIALS,
-    mode: str = defaults.FIND_BATCH_SIZE_MODE,
     steps_per_trial: int = defaults.FIND_BATCH_SIZE_STEPS_PER_TRIAL,
 ) -> None:
     """Computes maximum or optimal batch size.
 
     This sets the batch size on the datamodule, and if necessary, the
     gradient accumulation steps on the trainer.
 
+    With the max method, batch size is doubled until a memory error, thus
+    finding the largest batch size that fits in memory.
+
+    With the optimal method, batch size is doubled until a memory error or
+    once we confirm that the desired batch size fits in memory (whichever
+    comes first).
+
     Args:
         method (str): one of "max" (find the maximum batch size) or "opt" (find
             the "optimal" batch size, using the gradient accumulation trick if
             necessary.)
         trainer (pl.Trainer).
         model (models.BaseEncoderDecoder).
         datamodule (data.DataModule).
-        max_trials (int, optional): maximum number of increases in batch size
```

Comment:
Shouldn't we still support this? I thought your change enables a specific use-case, not that it fully replaces what you had before

Reply:
This is meaningless in `opt` mode and in `max` mode the default is fine (had I designed this I probably would have made this run forever but it is going to hit an OOM or get bigger than the training set long before 2^25...)

Reply:
Ok, so we basically think it is not a useful feature and therefore don't expose it.

Reply:
Yeah, I thought it would be useful, but there are two factors:

1. The time to run trials is linaer (or maybe a little worse than linear, IMO) with the size of the batches, so binsearch spends ln(n) time with the biggest, slowest batches.
2. When it settles on a max size it is going to have gotten lucky with having drawn three smaller batches.

I am still exposing the number of batches to draw per trial so if you want something more precise you can do it...

Part of me thinks I should just write my own scaler from scratch---it's not terribly complex as can be seen from the PTL impl.